reflex/compiler/compiler.py

```diff
@@ -111,14 +112,34 @@ def _compile_page(
     imports = utils.compile_imports(imports)
 
     # Compile the code to render the component.
-    return templates.PAGE.render(
+    template_obj = templates.PAGE.render(
         imports=imports,
         dynamic_imports=component.get_dynamic_imports(),
         custom_codes=component.get_custom_code(),
         state_name=state.get_name(),
         hooks=component.get_hooks(),
         render=component.render(),
     )
+    template_obj = _compile_conditional_rendering(template_obj)
+    return template_obj
+
+def _compile_conditional_rendering(template_obj: str) -> str:
+    """Update generated template to add required conditional rendering
+       for elements in the template
+
+    Args:
+        template_obj: Template to be modified
+
+    Returns:
+        The template after relevant modifications
+    """
+    pattern = re.compile(
+        r'(<DataTableGrid[^>]*\bdata\s*=\s*\{\s*\[\s*\]\s*\}[^>]*\/>)'
+    )
+    replacement = r'{ state.is_hydrated ? \1 : null }'
```

Comment:
the state var in the page is not always called `state`, the name is based on the class name that inherits from `rx.State`

Reply:
I agree that it would be much better for this logic to be encapsulated. Reviewing your code and I'm having difficulty understanding where or how the incoming data prop is set to type Var. On my end I'm only seeing it as type list when setting up the test application to match the conditions from #1830

Reply:
`data` would be a `Var` if it was coming from the state

```python
class State(rx.State):
    data: list[list[int]] = []

...

rx.data_table(data=State.data, columns=["A", "B"])
```

In this case, the `data` prop would be a `Var`. You can see this if you put a breakpoint in the `DataTable.create` method.

That said, I haven't attempted to repro this locally, so it's possible I might be misunderstanding the root cause of #1830